src/libgit2/pack.c

```diff
@@ -1525,6 +1525,14 @@ static int pack_entry_find_offset(
 	if (p->index_version > 1) {
 		level1_ofs += 2;
 		index += 8;
+
+		if ((int)short_oid->id[0] + 2 >= p->index_map.len) {
```

Comment:
I think that you need to upcast this to a `size_t` to be able to compare against the `len` (which is a `size_t` itself).